hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeyRequest.java

```diff
@@ -174,8 +174,11 @@ protected KeyArgs resolveBucketAndCheckOpenKeyAcls(KeyArgs keyArgs,
     return resolvedArgs;
   }
 
-  protected void validateKeyName(OzoneManager ozoneManager, String keyName)
-      throws OMException {
+  protected void validateKeyName(OzoneManager ozoneManager, KeyArgs keyArgs,
+      String keyName) throws OMException {
+    if (keyArgs != null && keyArgs.getKeyName() != null) {
```

Comment:
Shouldn't this check be on keyName? e.g. for the key rename operation, we need to verify that new name doesn't start with snapshot reserved keyword.

Reply:
There is already a check for keyName in the logic of OmUtils#validateKeyName().
HddsClientUtils.verifyKeyName():
![image](https://github.com/apache/ozone/assets/6416939/dd02639b-e434-4609-9068-1d06b3ec51c4)

Therefore, there is no need to check keyName shown here.

Reply:
Sorry if I was not clear earlier. I meant `OmUtils.verifyKeyNameWithSnapshotReservedWord()` should be called on `keyName` not `keyArgs.getKeyName()`. Because for `renameKey`, we should be validating that `toKeyName` doesn't contains snapshot reserved key word `.snapshot`.

Reply:
If that's the case, it seems like there's still something to discuss.
For OMKeyRenameRequest:
`
OmUtils.verifyKeyNameWithSnapshotReservedWord(renameKeyRequest.getToKeyName());
`
And for other types:
`
OmUtils.verifyKeyNameWithSnapshotReservedWord(keyArgs.getKeyName());
`
is that so?
@errose28 , what do you think?

Reply:
I think it was a miss or bug that `OmUtils.verifyKeyNameWithSnapshotReservedWord(renameKeyRequest.getToKeyName());` check was not added for rename.
This helper function should be:
```
  protected void validateKeyName(OzoneManager ozoneManager, String keyName) throws OMException {
    OmUtils.verifyKeyNameWithSnapshotReservedWord(keyName);
    final boolean checkKeyNameEnabled = ozoneManager.getConfiguration()
            .getBoolean(OMConfigKeys.OZONE_OM_KEYNAME_CHARACTER_CHECK_ENABLED_KEY,
                    OMConfigKeys.OZONE_OM_KEYNAME_CHARACTER_CHECK_ENABLED_DEFAULT);
    if (checkKeyNameEnabled) {
      OmUtils.validateKeyName(keyName);
    }
  }
```

In the end, we have to make sure the final key and dir name should be valid. For create request, final keyName is `keyArgs.getKeyName()` and for rename it is `renameKeyRequest.getToKeyName()`.